### DISCUSSION TITLE
Receive a list of available docs pages

### DISCUSSION THREAD

**Discussion by alihan-acn** [facebook/docusaurus#8574] [Q&A] [answered]
Hi,

I'm trying to create my frontpage to include an overview of all available pages with links to them. But I cannot find any way to receive the list of pages. I've looked at the `useDocusaurusContext` hook but it doesn't seem as it is included anywhere.

Is there any way to receive the actual available pages as an object? (The ones that are shown on http://.../docs/)

**Comment by danielleroux** [comment]
Dont know if it the best solution:
```javascript
  const pluginData = usePluginData('docusaurus-plugin-content-docs');
  console.log(pluginData.versions[0].docs);
```

**Reply by alihan-acn** [reply]
Hi @danielleroux,

I did consider a similar approach but it feels to me as this is not really what's intended, and I would love to avoid anything that can become deprecated in the near future - which in my optic this could seem to become. 

But if there is no other option, I might prompt for this. Thanks for sharing

**Comment by slorber** [comment, marked as answer]
`.docusaurus/routes.js` can be imported as `import routes from '@generated/routes';`

Our debug plugin uses it so it's unlikely we remove this feature without proposing an alternative:
https://docusaurus.io/__docusaurus/debug/routes

If you are looking for "reverse linking" to know which page links to which page, we don't support that atm but we probably want to add this. 

Might be related: https://github.com/facebook/docusaurus/discussions/8217 https://docusaurus.io/feature-requests/p/backlinks-or-pages-that-link-here

**Reply by alihan-acn** [reply]
Hi @slorber ,

This is perfect for retrieving the list of available routes with their corresponding urls. **It answers my question perfectly.** 

I do however have a different question after working a bit more with it. I was hoping to somehow access the meta data which resides within my markdown files, do you know any way to do that?

For example, if I have the following meta head in my markdown:
```
---
id: Test
title: Test Title
custom_var: Hello
---
```

I would like to retrieve a list of all my pages (like routes does) **and** the meta variables that exist within each of them. A solution could perhaps look something like the following json object;
```
[
...,
  {
      "path": "/docs/intro",
      "exact": true,
      "sidebar": "tutorialSidebar",
      "data": { id: 'test', title: 'Test Title', custom_var: 'Hello' },
  },
...
]
```

**Reply by slorber** [reply]
Unfortunately, it's not possible because all the pages do not have access to all the site's data. Some sites are pretty big and making this data available globally by default doesn't make sense and would make all your pages slower.

I have good hope that React Server Components could serve as a convenient layer to "pick" data from a big JSON DB without impacting the perf of a site, but we are not there yet.

In the meantime the most you can do is access the data that is made available to a specific page. You can't access docs metadata on your landing page, but you can access some docs metadata on a doc specific page. For example: `import {useDocsSidebar} from '@docusaurus/theme-common/internal';` can be used if you are inside a docs React page, because this data is available in React context.

---

You can also try to make the frontmatter data you need available globally by using this "plugin wrapping" trick I shared here: https://github.com/facebook/docusaurus/issues/4492#issuecomment-805041391. It should be possible to read docs metadata on the nodejs site and add the data you want to global data yourself as a post contentLoaded step.

(but keep in mind that makes your site global data larger, ie your landing page could become slow if you don't pay attention and add a lot of global data)
